src/setEnvVars.ts

```diff
@@ -0,0 +1,60 @@
+import { getNextConfig } from './nextConfig';
+import { loadEnvConfig } from '@next/env';
+
+const Logger = {
+  info: () => {},
+  error: () => {},
+};
+
+let dotFile: Record<string, string> | undefined = undefined;
+export function loadDotFile({ nextRoot }: { nextRoot: string }) {
+  // Temporarily hide global process.env to prevent "@next/env" from mutating it
+  const currentProcessEnv = process.env;
+  process.env = {};
+  const { combinedEnv } = loadEnvConfig(nextRoot, undefined, Logger);
```

Comment:
Should we add an assertion to our test that info about loaded file is loaded?

Reply:
The issue with how `@next/env` works is that, since we cannot control logger output, a user would get a log every time `getPage` is run. I'm considering reverting back to the original custom implementation and load only load `.env.test.local` or `.ent.test`.

 `@next/env` is actually able to merge multiple files, but I'd like to keep it simple. How do you see it?
 
Here is where  `@next/env` generates its dotfile list:
 ```js
  const isTest = process.env.NODE_ENV === 'test'
  const mode = isTest ? 'test' : dev ? 'development' : 'production'

  const dotenvFiles = [
    `.env.${mode}.local`,
    // Don't include `.env.local` for `test` environment
    // since normally you expect tests to produce the same
    // results for everyone
    mode !== 'test' && `.env.local`,
    `.env.${mode}`,
    '.env',
  ].filter(Boolean) as string[]
 ```

Reply:
Hmm, I don't have enough context to confidently say what is better. However, I think we should in general aim to re-use `nextjs` parts where possible. And this seems to be a public API (we are not importing some `/dist/` folder modules)

Reply:
Yep, it's a `next` dependency (but probably we should list it among ours deps). It's meant to be used by a real application (eg. it automatically loads env vars to global `process.env`, logs each time it's fired, etc), therefore it behaves in a way that sometimes clashes with our flow. Also: I'd like to reuse Next.js parts.